runtime/compiler/control/CompilationThread.cpp

```diff
@@ -12662,8 +12663,9 @@ void J9Method_HT::onClassUnloading(J9ClassLoader *j9classLoader)
       HT_Entry *prev = NULL;
       while (entry)
          {
-         if (NULL == entry->_j9method
-            || J9_CLASS_FROM_METHOD(entry->_j9method)->classLoader == j9classLoader)
+         J9Class *clazz = J9_CLASS_FROM_METHOD(entry->_j9method);
+         if ( J9_ARE_ALL_BITS_SET(clazz->classLoader->gcFlags, 0x2)
+            || (J9CLASS_FLAGS(clazz) & J9AccClassDying) )
```

Comment:
It would be good to add a comment here that the class specific check is required to handle vm anonymous classes which have a lifecycle independent of their classloader.  I had to double check the code when looking for how anon classes are handled so being explicit here would help.

Same with the code above

Reply:
I agree. Thanks!
See: 72c265fb0373f36991c94532768c471b29665784